net-libs/libwebsockets/libwebsockets-1.7.7.ebuild

```diff
@@ -0,0 +1,43 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit cmake-utils
+
+DESCRIPTION="canonical libwebsockets.org websocket library"
+HOMEPAGE="https://libwebsockets.org/"
+SRC_URI="https://github.com/warmcat/libwebsockets/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+http2 +server +ssl +zlib client ipv6 libev libressl libuv static-libs"
+
+RDEPEND="
+	libev? ( dev-libs/libev )
+	libuv? ( dev-libs/libuv )
+	ssl? (
+		!libressl? ( dev-libs/openssl:0= )
+		libressl?  ( dev-libs/libressl:0= )
+		)
+	zlib?  ( sys-libs/zlib )
```

Comment:
```
$ USE="-zlib" ebuild libwebsockets-1.7.7.ebuild configure
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.8") 
zlib include dirs: /usr/include
zlib libraries: /usr/lib64/libz.so
```
Disabling zlib makes no difference, the build system still performs detection to find it. I've passed:
```
-DLWS_WITH_ZLIB=$(usex zlib)
```
to see if it makes a difference but then if fails with:
```
CMake Error at CMakeLists.txt:138 (message):
  zlib is required for extensions.
```

I do not know the package well enough to say what those extensions are. If you think they should be included no matter what, then you can drop the zlib use flag and include sys-libs/zlib in RDEPEND unconditionally.